clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -778,7 +737,7 @@ public void testBuffersDereferencedOnClose(Args args) {
         int iterations =  0;
         while (iterations++ < 100) {
             buffer.rewind();
-            MemoryRecordsBuilder builder = new MemoryRecordsBuilder(buffer, RecordBatch.MAGIC_VALUE_V2, args.compressionType,
+            MemoryRecordsBuilder builder = new MemoryRecordsBuilder(buffer, args.magic, args.compressionType,
                     TimestampType.CREATE_TIME, 0L, 0L, RecordBatch.NO_PRODUCER_ID,
                     RecordBatch.NO_PRODUCER_EPOCH, RecordBatch.NO_SEQUENCE, false, false,
                     RecordBatch.NO_PARTITION_LEADER_EPOCH, 0);
```

Comment:
This is unrelated to this PR. However, could you replace `new byte[0]` by `key` (line#744)?

Reply:
@chia7712  @ijuma  I have fixed  it .  Thanks for both of you to review.